tensorflow/compiler/tf2tensorrt/utils/trt_shape_optimization_profiles.h

```diff
@@ -104,6 +115,16 @@ struct OptimizationProfileConfig {
   }
 };
 
+// Optimization profile generation strategies.
+//
+// The kUnitTest option is only used for unit testing to avoid TRT bug that
+// removes redundant input shape values (3153064).
+enum class ProfileStrategy {
+  kImplicitBatchModeCompatible,
+  kOptimal,
+  kUnitTest
```

Comment:
Sorry that I need to bring back a suggestion I made before about removing kUnitTest.
Does the bug we are trying to workaround using kUnitTest essentially mean that the current kImplicitBatchModeCompatible mode is buggy and we can't enable it?
What would be the problem if we (1) do not add kUnitTest, (2) instead add the workaround to kImplicitBatchModeCompatible (and add a TODO to remove the workaround when the NVIDIA bug is fixed)?

Reply:
This is the solution I suggested in email.
(1) remove kUnitTest. two justifications:
    (a) With kUnitTest, we are now only testing kUnitTest and don't test kImplicitModeCompatible at all.
    (b) It is hard to justify that we add stuff in the implementation just for the tests to pass.
         One may argue why can't we change the test, in this case, to avoid using a Shape tensor with d[0] =1 but instead d[0] = 2 or whatever to avoid hitting the bug

(2) "adjust all other profile strategies" with the following:
       If an input shape is 1D (shape tensors are 1D) and min==max then set max  to min+1.
       Add a TODO to such code block, to explain this is to workaround a TensorRT bug and the bug number.